osu.Game/Overlays/Rankings/RankingsSortTabControl.cs

```diff
@@ -1,19 +1,42 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using osu.Framework.Localisation;
+using osu.Game.Resources.Localisation.Web;
+
 namespace osu.Game.Overlays.Rankings
 {
     public class RankingsSortTabControl : OverlaySortTabControl<RankingsSortCriteria>
     {
         public RankingsSortTabControl()
         {
-            Title = "Show";
+            Title = RankingsStrings.FilterTitle;
         }
     }
 
+    [LocalisableEnum(typeof(RankingsSortCriteriaEnumLocalisationMapper))]
     public enum RankingsSortCriteria
     {
         All,
         Friends
     }
+
+    public class RankingsSortCriteriaEnumLocalisationMapper : EnumLocalisationMapper<RankingsSortCriteria>
```

Comment:
I guess this is something which was probably raised earlier, but going forward having a `LocalisableDescription` attribute (or something analague to this) to use in small enums instead of `EnumLocalisationMappers` to provide localisable descriptions may be better (and also would reduce  a bit the boilerplate, altough there are plans to provide code fixes to autogen the enum mappers iirc)

Reply:
The problem is that you can’t use LocalisableString types in attributes, only C# primitives. So you’re locked to using strings and `nameof(X)` here.

I’ll work on a generator (there’s an issue in ppy/osu-localisation-analyser for it).

Another option is to maybe also have the user type out `[LocalisableDescription(nameof(X.Y.Z))]` and have the analyser automatically split it out to `[LocalisableDescription(nameof(X), nameof(Y), nameof(Z))]`…

Reply:
I was thinking of something analogue to using class property names as lookup strings with nameof() and then resolve with reflection but in the end i'm fine with both propositions. I'll try to work out a framework PR with the mentionned approach to see how things look.

Reply:
I agree with having something like this, shortening the localisation procedure to this pretty simple way:

![CleanShot 2021-07-30 at 19 37 19](https://user-images.githubusercontent.com/22781491/127684593-9de3ba5a-1637-446b-8bc7-1f12f5ea36a6.png)

Reply:
Reflection isn't free. That sort of solution will likely be orders of magnitude slower.

If tooling can be improved to the point that the mappers can be automatically generated I see no real reason to switch.

Reply:
Reflection will be used any way you go for it if the link between the enum and the localisation strings is attribute-based.

I've benchmarked this and here's the results:

`EnumLocalisationMapper` way (constructs mapper, retrieves, and calls `Map` method, all via reflection):
```
|                    Method |     Mean |     Error |    StdDev |
|-------------------------- |---------:|----------:|----------:|
| GetLocalisableDescription | 5.726 us | 0.0181 us | 0.0151 us |
```
```
|                    Method | Times |        Mean |      Error |       StdDev |      Median |
|-------------------------- |------ |------------:|-----------:|-------------:|------------:|
| GetLocalisableDescription |    10 |    74.61 us |   1.479 us |     2.814 us |    73.92 us |
| GetLocalisableDescription |   100 |   849.18 us |  59.765 us |   169.545 us |   766.38 us |
| GetLocalisableDescription |  1000 | 9,579.63 us | 626.682 us | 1,837.951 us | 8,968.16 us |
```

`LocalisableDescription` way (retrieves member by specified name from specified declaring type, and gets value, all via reflection):
```
|                    Method |     Mean |     Error |    StdDev |
|-------------------------- |---------:|----------:|----------:|
| GetLocalisableDescription | 3.600 us | 0.0684 us | 0.1753 us |
```
```
|                    Method | Times |        Mean |      Error |     StdDev |      Median |
|-------------------------- |------ |------------:|-----------:|-----------:|------------:|
| GetLocalisableDescription |    10 |    30.30 us |   1.735 us |   4.779 us |    28.52 us |
| GetLocalisableDescription |   100 |   334.53 us |   6.674 us |   6.555 us |   333.23 us |
| GetLocalisableDescription |  1000 | 3,674.15 us | 188.477 us | 549.797 us | 3,477.65 us |
```

Showing that the `LocalisableDescription` way is *equally* performant, requiring less logic than the mapper way in usages, and can be used anywhere where `Description` is used.

I'll get into PR'ing the changes I locally have for supporting this. Further discussion can go there.

Reply:
Please make sure to include your benchmark code when you do so that it can be verified without reconstructing in its entirety.

Reply:
```csharp
    public class BenchmarkLocalisableDescription
    {
        [Benchmark]
        public LocalisableString GetLocalisableDescription() => TestLocalisableEnum.One.GetLocalisableDescription();
    }
```

```csharp
    public class BenchmarkLocalisableDescription
    {
        [Benchmark]
        public LocalisableString[] GetLocalisableDescription()
        {
            var array = new LocalisableString[Times];

            for (int i = 0; i < Times; i++)
                array[i] = TestLocalisableEnum.One.GetLocalisableDescription();

            return array;
        }
    }
```

(there was obviously no point in having separate code for "once" and "multiple times" benchmark, but didn't bother re-running another one with params)